metricbeat/module/windows/perfmon/pdh_query_windows.go

```diff
@@ -0,0 +1,208 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// +build windows
+
+package perfmon
+
+import (
+	"regexp"
+	"syscall"
+	"unsafe"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	instanceNameRegexp = regexp.MustCompile(`.*\((.*)\).*`)
+	objectNameRegexp   = regexp.MustCompile(`(?:^\\\\[^\\]+\\|^\\)([^\\]+)`)
+)
+
+// Counter object will contain the handle and format of each performance counter.
+type Counter struct {
+	handle       PdhCounterHandle
+	format       PdhCounterFormat
+	instanceName string
+}
+
+// Query contains the pdh.
+type Query struct {
+	handle   PdhQueryHandle
+	counters map[string]*Counter
+}
+
+// CounterValue contains the performance counter values.
+type CounterValue struct {
+	Instance    string
+	Measurement interface{}
+	Err         error
+}
+
+// Open creates a new query.
+func (q *Query) Open() error {
+	h, err := PdhOpenQuery("", 0)
+	if err != nil {
+		return err
+	}
+	q.handle = h
+	q.counters = make(map[string]*Counter)
+	return nil
+}
+
+// AddCounter adds the specified counter to the query.
+func (q *Query) AddCounter(counterPath string, counter CounterConfig, wildcard bool) error {
+	if _, found := q.counters[counterPath]; found {
+		return errors.Errorf("Counter %s has been already added", counterPath)
```

Comment:
Error messages should not be capitalized. https://github.com/golang/go/wiki/CodeReviewComments#error-strings